utils/firestore.js

```diff
@@ -2530,91 +2721,135 @@ const processParticipantHomeMouthwashKitData = (record, printLabel) => {
     connect_id: record['Connect_ID'],
     };
 
-    return (!hasMouthwash && printLabel) || (hasMouthwash && !printLabel)
+    return printLabel || hasMouthwash
         ? processedRecord
         : [];
 }
 
 const assignKitToParticipant = async (data) => {
+    let toReturn;
```

Comment:
Since this process has caused so many issues in the past, how about using a 'kitAssignmentResult' or 'kitAssignmentResultObj' variable, so we have specific messaging when something goes wrong? The transaction you've implemented should be great here, and specific messaging would really tighten up the unknown errors.

```
let kitAssignmentResult = { success: true, message: "Kit assigned successfully" };

//...
if (kitsWithDuplicateReturnTrackingNumbers.size > 0) {
    assignmentResult = {
        success: false,
        message: "Duplicate return tracking number found",
        duplicateReturnTrackingNumber: data[supplyKitTrackingNum]
    };
    return;
}

//...

if (otherKitsUsingSupplyKitTrackingNumber.size > 1) {
    assignmentResult = {
        success: false,
        message: "Multiple kits found with the same supply kit tracking number",
        supplyKitTrackingNumber: data[supplyKitTrackingNum]
    };
    return;
}

//...etc.
```